libp2p/wire.nim

```diff
@@ -33,7 +33,7 @@ proc initTAddress*(ma: MultiAddress): MaResult[TransportAddress] {.
   ## Initialize ``TransportAddress`` with MultiAddress ``ma``.
   ##
   ## MultiAddress must be wire address, e.g. ``{IP4, IP6, UNIX}/{TCP, UDP}``.
-  if TRANSPMA.match(ma):
+  if TRANSPMA.matchPartial(ma):
```

Comment:
What the purpose of doing this change?

Reply:
To fix directchat
and to allow such addresses `/ip4/127.0.0.1/tcp/55505/ipfs/QmRop5a5niZ9rpkHYEsrctyZckgfL3JKXzPQLsaT8fkFao` else would not work

Reply:
Hmm, yes it's indeed weird that this broke, there have been no changes to this functionality AFAIK. Perhaps there is something else going on? On the other hand, I also wonder what are the implications of this change - @cheatfate you worked on this primitives and clue whats going on?

Reply:
This is very weird, because `directchat.nim` do not use this procedures directly. From my point of view `wire.nim` should accept only addresses without any IPFS information, because `wire.nim` is about TCP,UDP,UNIX over IPv4/IPv6 networking but nothing else. Its the glue between `libp2p` and `chronos` transports.

So for example when you change `match` to `matchPartial`, then
addresses like `/ip4/127.0.0.1/tcp/80/http` become accepted, but this addresses should not be accepted by TCP transport it should be accepted by more high level type of transport which recognize `http` part of path too... So its why i have made here strict comparison. So if you need to use this procedure you should split your MultiAddress into parts and this will confirm that you understand what you are doing.

So you should fix `directchat.nim` in some other place, or even fix `directchat.nim` itself to split addresses, because it should not rely on `wire.nim` procedures (it even not imports this file).

Reply:
Ah, right - yes I completely agree, this is something that needs to happen before it makes it to wire.

Reply:
This example was written with that wrong assumption in mind tho.
It's easy to understand that.
Should we rewrite it?

Reply:
Yeah, weird that it's like that, it uses to work but maybe it was wrong all along. Let's fix it to properly reflect how it should work.